src/types.test.ts

```diff
@@ -836,158 +836,6 @@ describe('c.var with chaining - test only types', () => {
       return c.json(0)
     })
 
-    // app.get('/', handler...)
-
-    new Hono().get('/', mw1).get('/', (c) => {
```

Comment:
As you are doing, I think these tests should be removed.

[As I mentioned](https://github.com/honojs/hono/issues/2019#issuecomment-1900254498), the correct behavior is for these tests to fail, since the chained return value app should not inherit the env defined in `mw1`.